policyengine_api/utils/logger.py

```diff
@@ -0,0 +1,116 @@
+import logging
+import sys
+from google.cloud import logging as cloud_logging
+from datetime import datetime
+from pathlib import Path
+import os
+
+
+class Logger:
+    def __init__(
+        self,
+        dir: str = "logs",
+        name: str = "api_main",
+        id: str = None,
+        log_to_cloud: bool = True,
+    ):
+        """
+        Initialize standard logger
+
+        Filepath:
+        dir/name_id.log
+
+        Args:
+            dir (str): Directory to store log files (defaults to "logs")
+            name (str): Name of the logger (defaults to "api_main")
+            id (str): ID to append to log file name; if not provided, will use current timestamp
+            log_to_cloud (bool): Whether to log to Google Cloud Logging
+        """
+        # Generate three parts of storage path
+        self.dir = Path(dir)
+        self.name = name
+        # Don't pass datetime to initializer, otherwise time is equal to when class is created
+        self.id = (
+            id if id is not None else datetime.now().strftime("%Y%m%d_%H%M%S")
+        )
+
+        self.logger = logging.getLogger(self.name)
+        self.logger.setLevel(logging.INFO)
+
+        # Create log directory if it doesn't exist
+        try:
+            self.dir.mkdir(parents=True, exist_ok=True)
+        except Exception as e:
```

Comment:
**question, blocking** - who/what is this for? Google app engine looks at [either stdout/stderr or the cloud logging API](https://cloud.google.com/appengine/docs/standard/writing-application-logs?tab=python#top).

For that reason I'm surprised we write this by default even in the cloud.

**suggestion** I would not do this by default I would only do it in dev mode. In the context of the suggestion above I would pass a different config in dev than in prod.